src/App.tsx

```diff
@@ -1,24 +1,184 @@
 import React from 'react';
 import './App.scss';
 
-const goodsFromServer: string[] = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
-
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+type State = {
+  selectedGood: string;
+};
+
+export class App extends React.Component<{}, State> {
+  goodsFromServer = [
+    'Dumplings',
+    'Carrot',
+    'Eggs',
+    'Ice cream',
+    'Apple',
+    'Bread',
+    'Fish',
+    'Honey',
+    'Jam',
+    'Garlic',
+  ];
+
+  state = {
+    selectedGood: '',
+  };
+
+  componentDidMount() {
+  }
+
+  componentDidUpdate(prevState: State) {
+    if (prevState.selectedGood !== this.state.selectedGood) {
+      const initialText = document.querySelector('.initial_text');
+      const stateLength = this.state.selectedGood.split(',').length;
+
+      if (initialText !== null) {
+        if (stateLength === 1 && this.state.selectedGood !== '') {
+          initialText.textContent = `${this.state.selectedGood.replaceAll('_', ' ')} is selected`;
+        } else if (this.state.selectedGood === '') {
+          initialText.textContent = 'No goods selected';
+        } else {
+          initialText.textContent = `${this.state.selectedGood.replaceAll('_', ' ')} are selected`;
+        }
+      }
+    }
+  }
+
+  addItem = (item: string, event: React.MouseEvent) => {
+    const nonActiveItem = document.querySelector(`.${item}`);
+    const clickedButton = event.target;
+    const buttons = document.querySelectorAll('.button');
+    const buttonX: HTMLElement | null = document.querySelector('.buttonX');
+
+    if (buttonX !== null) {
+      buttonX.hidden = false;
+    }
+
+    buttons.forEach(button => {
+      if (button === clickedButton) {
+        button.classList.add('hidden');
```

Comment:
https://www.npmjs.com/package/classnames
don't add classes manually to elements, use library for that